### PR TITLE
Add subscription in vhdSchema to support that vhd is in another subsc…

### PR DESCRIPTION
**Pull request by lubaihua33**
Add subscription in vhdSchema to support that vhd is in another subscription.

This PR doesn't contain the scenario Sig transformer. It will in another PR.

### CODE REVIEW

File: lisa/sut_orchestrator/azure/common.py [hunk 1]
```diff
@@ -2225,9 +2236,12 @@ def get_deployable_vhd_path(
     """
     matches = SAS_URL_PATTERN.match(vhd_path)
     if not matches:
-        vhd_details = get_vhd_details(platform, vhd_path)
+        vhd_details = get_vhd_details(platform, vhd_path, vhd_subscription)
         vhd_location = vhd_details["location"]
-        if location == vhd_location:
+        if (
+            location == vhd_location
+            and vhd_details["subscription"] == platform.subscription_id
```
Comment (LiliDeng) [inline lisa/sut_orchestrator/azure/common.py:2242]: vhd_details may not contain subscription

File: lisa/sut_orchestrator/azure/common.py [hunk 2]
```diff
@@ -2216,7 +2223,11 @@ def _generate_sas_token_for_vhd(
 
 @lru_cache(maxsize=10)  # noqa: B019
 def get_deployable_vhd_path(
-    platform: "AzurePlatform", vhd_path: str, location: str, log: Logger
+    platform: "AzurePlatform",
+    vhd_path: str,
+    location: str,
+    log: Logger,
+    vhd_subscription: str = "",
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/common.py]: name it `subscription`.

File: lisa/sut_orchestrator/azure/common.py (hunk 1 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/common.py:2242]: Have added.

File: lisa/sut_orchestrator/azure/common.py (hunk 2 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/common.py]: Done

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**